validator/coordinator/src/main/java/tech/pegasys/teku/validator/coordinator/BlockFactory.java

```diff
@@ -83,6 +83,22 @@ public BlockFactory(
     this.spec = spec;
   }
 
+  public void prepareExecutionPayload(
+      final Optional<BeaconState> maybeCurrentSlotState, UInt64 payloadId) {
+    if (maybeCurrentSlotState.isEmpty()) return;
+    final BeaconState currentState = maybeCurrentSlotState.get();
+    if (currentState instanceof BeaconStateMerge) {
```

Comment:
The pattern we are sticking to is `BeaconState.toVersionMerge()`. In the future `BeaconStateMerge` inheritance could be removed.

Reply:
oky! updating this

Reply:
updated but still not a clean solution